kivy/uix/image.py

```diff
@@ -400,3 +401,74 @@ def reload(self):
             Loader.remove_from_cache(source)
 
         super(AsyncImage, self).reload()
+
+
+class Base64Image(Image):
+
+    __events__ = ('on_error', 'on_load')
+
+    def __init__(self, **kwargs):
+        self._coreimage = None
+        super(Base64Image, self).__init__(**kwargs)
+        self.fbind('source', self._load_source)
+        if self.source:
+            self._load_source()
+        self.on_anim_delay(self, kwargs.get('anim_delay', .25))
+
+    def _load_source(self, *args):
+        source = self.source
+        if not source:
+            if self._coreimage is not None:
+                self._coreimage.unbind(on_texture=self._on_tex_change)
+                self._coreimage.unbind(on_load=self._on_source_load)
+            self.texture = None
+            self._coreimage = None
+        else:
+            try:
+                image64 = source.split(';base64,')[1]
+                type64 = source.split('data:image/')[1] \
+                                .split(';base64,')[0]
+            except:
+                raise Exception('Wrong type of base64 image.')
+            base64l = base64.b64decode(image64)
+            self._coreimage = image = CoreImage(
+                BytesIO(base64l), ext = type64,
+                nocache=self.nocache, mipmap=self.mipmap,
+                anim_delay=self.anim_delay)
+
+            image.bind(on_load=self._on_source_load)
+            image.bind(on_error=self._on_source_error)
+            image.bind(on_texture=self._on_tex_change)
+            self.texture = image.texture
+
+    def _on_source_load(self, value):
+        image = self._coreimage.image
+        if not image:
```

Comment:
minor: do you mean `if image is None:`?

Reply:
This is the part I do not really understand how these rules are working. Someone needs to cure that!

Reply:
OK, it was minor anyway, but I was talking about the Pythonic to check for `None` https://www.python.org/dev/peps/pep-0008/#programming-recommendations

Reply:
Aaa okay!

Reply:
addressed in daab224